src/docs/memory-management.md

```diff
@@ -0,0 +1,103 @@
+# Memory management
+
+This section outlines the native memory management approach for projected value types, focusing on when and how to retain and release ref counters. The goal is to ensure that native memory is handled correctly across Swift and C# boundaries.
+
+## Requirements
+ - Projected types shall call the value witness table functions to retain and release owned ref types when required
+ - Projected types shall not invoke retain or release functions when implicit copies of ref types are created on the C# side
+
+## Memory ownership
```

Comment:
I would make it explicit that this describes Swift behavior - and only Swift behavior.